File: deprecated/apps/agents/desktop/telemetry/sysfs_other.go
//go:build !linux && !windows

package telemetry

// readCPUTimes is a stub for unsupported platforms.
func readCPUTimes() (idle, total uint64) {
	return 0, 0
}

// calculateCPUUsage is a stub for unsupported platforms.
func calculateCPUUsage(prevIdle, prevTotal, currIdle, currTotal uint64) float64 {
	return -1
}

// readCPUTemp is a stub for unsupported platforms.
func readCPUTemp() float64 {
	return -1
}

// readGPUUsage is a stub for unsupported platforms.
func readGPUUsage() float64 {
	return -1
}

// readGPUTemp is a stub for unsupported platforms.
func readGPUTemp() float64 {
	return -1
}

// readMemInfo is a stub for unsupported platforms.
func readMemInfo() (total, available int64) {
	return -1, -1
}

// readBattery is a stub for unsupported platforms.
func readBattery() (capacity int, status string) {
	return -1, "Unknown"
}

// readCPUFreq is a stub for unsupported platforms.
func readCPUFreq() float64 {
	return -1
}

// readGPUFreq is a stub for unsupported platforms.
func readGPUFreq() float64 {
	return -1
}

// readPowerInfo is a stub for unsupported platforms.
func readPowerInfo() (tdpWatts, powerWatts float64) {
	return -1, -1
}

// readFanSpeed is a stub for unsupported platforms.
func readFanSpeed() int {
	return -1
}

File: crates/telemetry/src/sysfs_other.rs
//! Stub metric readers for unsupported platforms.

pub fn read_cpu_times() -> (u64, u64) {
    (0, 0)
}
pub fn read_cpu_temp() -> f64 {
    -1.0
}
pub fn read_cpu_freq() -> f64 {
    -1.0
}
pub fn read_gpu_usage() -> f64 {
    -1.0
}
pub fn read_gpu_temp() -> f64 {
    -1.0
}
pub fn read_gpu_freq() -> f64 {
    -1.0
}
pub fn read_gpu_mem_freq() -> f64 {
    -1.0
}
pub fn read_vram() -> (i64, i64) {
    (-1, -1)
}
pub fn read_mem_info() -> (i64, i64, i64, i64) {
    (-1, -1, -1, -1)
}
pub fn read_battery() -> (i32, String) {
    (-1, String::new())
}
pub fn read_power_info() -> (f64, f64) {
    (-1.0, -1.0)
}
pub fn read_fan_speed() -> i32 {
    -1
}
